cdap-app-templates/cdap-etl/cdap-data-pipeline-base/src/main/java/io/cdap/cdap/datapipeline/SmartWorkflow.java

```diff
@@ -427,6 +427,18 @@ public void initialize(WorkflowContext context) throws Exception {
     }
     spec = GSON.fromJson(context.getWorkflowSpecification().getProperty(Constants.PIPELINE_SPEC_KEY),
                          BatchPipelineSpec.class);
+
+    WorkflowContext workflowContext = getContext();
+    if (workflowContext.getDataTracer(PostAction.PLUGIN_TYPE).isEnabled()) {
+      for (StageSpec stageSpec : spec.getStages()) {
+        String pluginType = stageSpec.getPluginType();
+        if (pluginType.equals("condition")) {
+          WRAPPERLOGGER.error("Can not run pipeline with Conditional in preview mode.");
```

Comment:
Should remove this log message. Logging and then throwing an exception is usually not the right thing to do, as it often ends up double logging. Once here, then once by whatever ends up handling the exception.